src/assets/scripts/client/aircraft/Pilot/Pilot.js

```diff
@@ -145,6 +145,19 @@ export default class Pilot {
         readback.log = `${altitudeInstruction} ${readbackAltitude}${expediteReadback}`;
         readback.say = `${altitudeInstruction} ${altitudeVerbal}${expediteReadback}`;
 
+        if (!aircraftModel.model.isAbleToMaintainAltitude(altitude)) {
+            const verbalRequestedAltitude = radio_altitude(altitude);
+
+            readback.log = `unable to maintain ${altitude} due to performance`;
+            readback.say = `unable to maintain ${verbalRequestedAltitude} due to performance`;
+
+            this._mcp.setAltitudeFieldValue(aircraftModel.altitude);
```

Comment:
Once we move this block up, it should not be setting anything at all. Only returning early with a response that explains why the pilot decided not to follow the instruction.

Reply:
done